src/components/AboutVisualizationUnit/AboutVisualizationUnit.js

```diff
@@ -0,0 +1,263 @@
+import { useDataQuery, useDataMutation } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import {
+    Button,
+    CircularLoader,
+    IconChevronDown24,
+    IconChevronUp24,
+    IconClock16,
+    IconShare16,
+    IconSubscribe24,
+    IconSubscribeOff24,
+    IconUser16,
+    IconView16,
+    colors,
+} from '@dhis2/ui'
+import cx from 'classnames'
+import moment from 'moment'
+import PropTypes from 'prop-types'
+import React, { useEffect, useMemo, useState } from 'react'
+import classes from './styles/AboutVisualizationUnit.module.css'
+
+const getQueries = type => ({
+    visualization: {
+        resource: type,
+        id: ({ id }) => id,
+        params: {
+            fields: 'id,displayDescription,created,createdBy[displayName],lastUpdated,subscribed,publicAccess,userAccesses[displayName,access],userGroupAccesses[displayName,access]',
+        },
+    },
+    dataStatistics: {
+        resource: 'dataStatistics/favorites',
+        id: ({ id }) => id,
+    },
+})
+
+const getSubscribeMutation = (type, id) => ({
+    resource: `${type}/${id}/subscriber`,
+    type: 'create',
+})
+
+const getUnsubscribeMutation = (type, id) => ({
+    resource: `${type}/${id}/subscriber`,
+    type: 'delete',
+})
+
+const AboutVisualizationUnit = ({ type, id }) => {
+    const [isExpanded, setIsExpanded] = useState(true)
+
+    const queries = useMemo(() => getQueries(type), [])
+
+    const {
+        data,
+        loading: dataIsLoading,
+        refetch,
+    } = useDataQuery(queries, {
+        lazy: true,
+    })
+
+    const subscribeMutation = useMemo(() => getSubscribeMutation(type, id), [])
+
+    const unsubscribeMutation = useMemo(
+        () => getUnsubscribeMutation(type, id),
+        []
+    )
+
+    const [subscribe, { loading: subscribeIsLoading }] = useDataMutation(
+        subscribeMutation,
+        {
+            onComplete: res => {
+                if (res.status === 'OK') {
+                    refetch({ id })
+                }
+            },
+        }
+    )
+
+    const [unsubscribe, { loading: unsubscribeIsLoading }] = useDataMutation(
+        unsubscribeMutation,
+        {
+            onComplete: res => {
+                if (res.status === 'OK') {
+                    refetch({ id })
+                }
+            },
+        }
+    )
+
+    useEffect(() => {
+        if (id) {
+            refetch({ id })
+        }
+    }, [type, id])
+
+    const getAccessLevelString = access => {
+        const accessMatch = access.match(/(?<accessLevel>rw?)/)
+
+        switch (accessMatch.groups.accessLevel) {
+            case 'r':
+                return i18n.t('view only')
+            case 'rw':
```

Comment:
feels like there's a quite a few of these strings being handled throughout the code. could these be made into consts instead? e.g. `const USER_ACCESS_READ = 'r'`? Feels like there could easily be a typo somewhere otherwise and I suppose all of this is quite sensitive...

Reply:
Added constants and rewritten the regexes to use them.